examples/tv-casting-app/android/App/app/src/main/jni/com/matter/casting/core/CastingPlayerDiscovery.java

```diff
@@ -0,0 +1,132 @@
+/*
+ *   Copyright (c) 2023 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.matter.casting.core;
+
+import android.util.Log;
+import com.matter.casting.support.MatterError;
+import java.util.List;
+
+/**
+ * The CastingPlayerDiscovery interface defines the API to control Matter Casting Player discovery
+ * over DNS-SD, and to collect discovery results. Discovery is centrally managed by the native C++
+ * layer in the Matter SDK. This class exposes native functions to add and remove a
+ * CastingPlayerChangeListener, which contains the C++ to Java callbacks for when Casting Players
+ * are discovered, updated, or lost from the network. This class is a singleton.
+ */
+public interface CastingPlayerDiscovery {
+
+  /**
+   * @return a list of Casting Players discovered during the current discovery session. This list is
+   *     cleared when discovery stops.
+   */
+  List<CastingPlayer> getCastingPlayers();
+
+  /**
+   * Starts Casting Players discovery or returns an error.
+   *
+   * @param discoveryTargetDeviceType the target device type to be discovered using DNS-SD. 35
+   *     represents device type of Matter Casting Player.
+   * @return a specific MatterError if the the operation failed or NO_ERROR if succeeded.
+   */
+  MatterError startDiscovery(int discoveryTargetDeviceType);
+
+  /**
+   * Stops Casting Players discovery or returns an error.
+   *
+   * @return a specific MatterError if the the operation failed or NO_ERROR if succeeded.
+   */
+  MatterError stopDiscovery();
+
+  /**
+   * Adds a CastingPlayerChangeListener instance to be used during discovery. The
+   * CastingPlayerChangeListener contains the C++ to Java callbacks for when Casting Players are
```

Comment:
```suggestion
   * CastingPlayerChangeListener defines handler methods for when Casting Players are
```

The caller of these APIs does not need to know how the implementation has been done (don't mention "C++ to Java")